### PR TITLE
Use `get-tsconfig` to parse a given tsconfig file

### LINKED ISSUE #20: Plugin does not resolve `extends` in `tsconfig.json`
Currently, it seems the `dev-server-esbuild` plugin does not resolve the `extends` field in `tsconfig.json`. 

I have a small repro here: https://github.com/peschee/wds-esbuild-test/tree/with-multiple-tsconfig

Given the following `tsconfig.base.json` file:

```json
{
    "compilerOptions": {
      "target": "es2020",
      "module": "nodenext",
      "lib": ["es2020", "dom", "dom.iterable"],
      "declaration": false,
      "strict": true,
      "noUnusedLocals": true,
      "noUnusedParameters": true,
      "noImplicitReturns": true,
      "noFallthroughCasesInSwitch": true,
      "noImplicitAny": true,
      "noImplicitThis": true,
      "moduleResolution": "nodenext",
      "allowSyntheticDefaultImports": true,
      "experimentalDecorators": true,
      "useDefineForClassFields": false,
      "forceConsistentCasingInFileNames": true,
      "noEmit": true,
      "skipLibCheck": true,
      "esModuleInterop": true
    },
    "include": ["src/**/*.ts", "test/**/*.ts"]
  }
```

and a `tsconfig.json` which just extends this:

```json
{
    "extends": "./tsconfig.base.json",
}
```

The `web-dev-server` cannot transform a simple `LitElement` component using decorators, even though in `tsconfig.base.json` we explicitly enable this through `"experimentalDecorators": true`.

```
Error while transforming src/components/app-shell/app-shell.ts: Transforming JavaScript decorators to the configured target environment ("es2022") is not supported yet
```

My hunch is that the problem is this line: https://github.com/modernweb-dev/dev-server-esbuild/blob/main/src/EsbuildPlugin.ts#L64

Where the TS config is just read using `fs.readFile` which obviously won't resolve the `extends` field.

It seems that `esbuild` can do this out of the box: https://github.com/peschee/esbuild-tsconfig

Probably also relates to https://github.com/modernweb-dev/web/issues/1921

### PR DESCRIPTION
**Pull request by peschee**
This PR fixes #20 by using `get-tsconfig`'s `parseTsconfig` method to return a fully resolved TS config.

### CODE REVIEW

File: src/EsbuildPlugin.ts [hunk 1]
```diff
@@ -61,7 +62,8 @@ export class EsbuildPlugin implements Plugin {
     this.config = config;
     this.logger = logger;
     if (this.esbuildConfig.tsconfig) {
-      this.tsconfigRaw = await promisify(fs.readFile)(this.esbuildConfig.tsconfig, 'utf8');
+      const parsedTsconfig = await parseTsconfig(this.esbuildConfig.tsconfig);     
+      this.tsconfigRaw = parsedTsconfig ? JSON.stringify(parsedTsconfig) : await promisify(fs.readFile)(this.esbuildConfig.tsconfig, 'utf8');
```
Comment (peschee) [inline src/EsbuildPlugin.ts]: For now, I have added this on top of the `fs.readFile` call, but the latter becomes redundant, so we could also drop it entirely.

Comment (peschee) [reply src/EsbuildPlugin.ts]: `parseTsconfig` already throws anyway when something goes wrong reading the file, so we can ditch `fs.readFile` here.

**Comment by thepassle** [comment]
@peschee can you run `npm run format` in the monorepo root? Probably prettier is complaining

**Comment by peschee** [comment]
@thepassle Done!

**Comment by thepassle** [comment]
@peschee published in `1.0.2`

**Pull request merged**